File: Wawaji-Client-iOS/Wawaji-Client-iOS/WawajiControllerCreator.m
//
//  WawajiControllerCreator.m
//  Wawaji-Client-iOS
//
//  Created by suleyu on 2017/12/2.
//  Copyright © 2017 Agora. All rights reserved.
//

#import "WawajiControllerCreator.h"
#import "LeidiWawajiController.h"
#import "LeyaoyaoWawajiController.h"
#import "ZinianWawajiController.h"

@implementation WawajiControllerCreator

+ (id<WawajiController>)getWawajiController:(WawajiManufacturer)manufacturer {
    id<WawajiController> wawajiController;
    
    switch (manufacturer) {
        case WawajiManufacturer_LeiDi:
            wawajiController = [[LeidiWawajiController alloc] init];
            break;
            
        case WawajiManufacturer_LeYaoYao:
            wawajiController = [[LeyaoyaoWawajiController alloc] init];
            break;
            
        case WawajiManufacturer_ZiNian:
            wawajiController = [[ZinianWawajiController alloc] init];
            break;
            
        default:
            wawajiController = nil;
            break;
    }
    
    return wawajiController;
}

@end

File: Wawaji-Client-iOS-Swift/Wawaji-Client/WawajiControllerCreator.swift
//
//  WawajiControllerCreator.swift
//  Wawaji-Client
//
//  Created by suleyu on 2017/12/1.
//  Copyright © 2017 Agora. All rights reserved.
//

import Foundation

 enum WawajiManufacturer {
    case LeYaoYao
    case ZiNian
 }

 class WawajiControllerCreator {

    class func getWawajiController(manufacturer: WawajiManufacturer) ->WawajiController {
        switch manufacturer {
        case .LeYaoYao:
            return LeyaoyaoWawajiController()

        case .ZiNian:
            return ZinianWawajiController()
        }
    }
 }
